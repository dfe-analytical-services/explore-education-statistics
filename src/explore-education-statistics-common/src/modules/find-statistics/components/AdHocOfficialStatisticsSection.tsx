import React from 'react';

const AdHocOfficialStatisticsSection = () => (
  <>
    <p>
      Ad hoc official statistics are one off publications that have been
      produced as far as possible in line with the Code of Practice for
      Statistics.
    </p>
    <p>This can be broadly interpreted to mean that these statistics are:</p>
    <ul>
      <li>managed impartially and objectively in the public interest</li>

      <li>meet identified user needs</li>

      <li>produced according to sound methods</li>

      <li>well explained and readily accessible</li>
    </ul>
    <p>
      Find out more about the standards we follow to produce these statistics
      through our{' '}
      <a href="https://www.gov.uk/government/publications/standards-for-official-statistics-published-by-the-department-for-education">
        Standards for official statistics published by DfE guidance
      </a>
      .
    </p>
  </>
);

export default AdHocOfficialStatisticsSection;
