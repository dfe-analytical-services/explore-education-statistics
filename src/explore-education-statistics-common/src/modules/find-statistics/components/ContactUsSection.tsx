import { Contact } from '@common/services/publicationService';
import React from 'react';

interface Props {
  publicationContact: Contact;
  publicationTitle: string;
}

const ContactUsSection = ({ publicationContact, publicationTitle }: Props) => {
  return (
    <>
      <p>
        If you have a specific enquiry about {publicationTitle} statistics and
        data:
      </p>
      <h4 className="govuk-heading-s govuk-!-margin-bottom-0">
        {publicationContact.teamName}
      </h4>
      <p className="govuk-!-margin-top-0">
        Email <br />
        <a href={`mailto:${publicationContact.teamEmail}`}>
          {publicationContact.teamEmail}
        </a>
      </p>
      <p>
        Telephone: {publicationContact.contactName} <br />{' '}
        {publicationContact.contactTelNo}
      </p>
      <h4 className="govuk-heading-s govuk-!-margin-bottom-0">Press office</h4>
      <p className="govuk-!-margin-top-0">If you have a media enquiry:</p>
      <p>
        Telephone <br />
        020 7783 8300
      </p>
      <h4 className="govuk-heading-s govuk-!-margin-bottom-0">
        Public enquiries
      </h4>
      <p className="govuk-!-margin-top-0">
        If you have a general enquiry about the Department for Education (DfE)
        or education:
      </p>
      <p>
        Telephone <br />
        037 0000 2288
      </p>
      <p>
        Opening times: <br />
        Monday to Friday from 9.30am to 5pm (excluding bank holidays)
      </p>
    </>
  );
};

export default ContactUsSection;
