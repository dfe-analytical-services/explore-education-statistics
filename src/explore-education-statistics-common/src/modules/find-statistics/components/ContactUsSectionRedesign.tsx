import { Contact } from '@common/services/publicationService';
import React from 'react';

export const contactUsNavItem = {
  id: 'contact-us-section',
  text: 'Contact us',
};

const ContactUsSection = ({
  sectionTitle = 'Contact us',
  publicationContact,
  publicationTitle,
}: {
  sectionTitle?: string;
  publicationContact: Contact;
  publicationTitle: string;
}) => {
  return (
    <section id="contact-us-section" data-page-section>
      <h2 id="contact-us">{sectionTitle}</h2>

      <p>
        If you have a specific enquiry about {publicationTitle} statistics and
        data:
      </p>

      <h3 className="govuk-heading-s govuk-!-margin-bottom-0">
        {publicationContact.teamName}
      </h3>

      <address className="govuk-!-margin-top-0">
        Email:{' '}
        <a href={`mailto:${publicationContact.teamEmail}`}>
          {publicationContact.teamEmail}
        </a>
        <br />
        Contact name: {publicationContact.contactName}
        {publicationContact.contactTelNo && (
          <>
            <br />
            Telephone: {publicationContact.contactTelNo}
          </>
        )}
      </address>

      <h3 className="govuk-heading-s govuk-!-margin-bottom-0">Press office</h3>

      <p className="govuk-!-margin-top-0 govuk-!-margin-bottom-0">
        If you have a media enquiry:
      </p>
      <p>Telephone: 020 7783 8300</p>

      <h3 className="govuk-heading-s govuk-!-margin-bottom-0">
        Public enquiries
      </h3>

      <p className="govuk-!-margin-top-0 govuk-!-margin-bottom-0">
        If you have a general enquiry about the Department for Education (DfE)
        or education:
      </p>
      <p>Telephone: 037 0000 2288</p>

      <h3 className="govuk-heading-s govuk-!-margin-bottom-0">Opening times</h3>
      <p className="govuk-!-margin-top-0">
        Monday to Friday from 9.30am to 5pm (excluding bank holidays)
      </p>
    </section>
  );
};

export default ContactUsSection;
