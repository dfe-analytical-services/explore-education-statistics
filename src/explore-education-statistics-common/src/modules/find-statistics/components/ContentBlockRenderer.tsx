import ContentHtml from '@common/components/ContentHtml';
import { ContentBlock } from '@common/services/types/blocks';
import { Dictionary } from '@common/types';
import {
  defaultSanitizeOptions,
  SanitizeHtmlOptions,
} from '@common/utils/sanitizeHtml';
import React, { useMemo } from 'react';
import ReactMarkdown from 'react-markdown';
import { GlossaryEntry } from '@common/services/types/glossary';

interface Props {
  block: ContentBlock;
  transformImageAttributes?: (
    attributes: Dictionary<string>,
  ) => Dictionary<string>;
  getGlossaryEntry?: (slug: string) => Promise<GlossaryEntry>;
  trackGlossaryLinks?: (glossaryEntrySlug: string) => void;
}

const ContentBlockRenderer = ({
  block,
  transformImageAttributes,
  getGlossaryEntry,
  trackGlossaryLinks,
}: Props) => {
  const { body = '', type } = block;

  const sanitizeOptions: SanitizeHtmlOptions = useMemo(() => {
    return {
      ...defaultSanitizeOptions,
      transformTags: {
        img: (tagName, attribs) => {
          return {
            tagName,
            attribs: transformImageAttributes
              ? transformImageAttributes(attribs)
              : attribs,
          };
        },
      },
    };
  }, [transformImageAttributes]);

  switch (type) {
    case 'MarkDownBlock':
      return <ReactMarkdown className="dfe-content" source={body} />;
    case 'HtmlBlock':
      return (
        <ContentHtml
          html={body}
          sanitizeOptions={sanitizeOptions}
          getGlossaryEntry={getGlossaryEntry}
          trackGlossaryLinks={trackGlossaryLinks}
        />
      );
    default:
      return null;
  }
};

export default ContentBlockRenderer;
