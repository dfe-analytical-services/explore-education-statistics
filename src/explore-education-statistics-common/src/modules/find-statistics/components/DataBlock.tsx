import LoadingSpinner from '@common/components/LoadingSpinner';
import Tabs from '@common/components/Tabs';
import TabsSection from '@common/components/TabsSection';
import ChartRenderer, {
  ChartRendererProps,
} from '@common/modules/find-statistics/components/ChartRenderer';

import DataSource from '@common/modules/find-statistics/components/DataSource';
import SummaryRenderer, {
  SummaryRendererProps,
} from '@common/modules/find-statistics/components/SummaryRenderer';
import TimePeriodDataTableRenderer, {
  Props as TableRendererProps,
} from '@common/modules/find-statistics/components/TimePeriodDataTableRenderer';
import DataBlockService, {
  DataBlockData,
  DataBlockRequest,
  DataBlockResponse,
} from '@common/services/dataBlockService';
import {
  Chart,
  DataQuery,
  Summary,
  Table,
} from '@common/services/publicationService';
import React, { Component, MouseEvent, ReactNode } from 'react';
import { parseMetaData } from '@common/modules/find-statistics/components/charts/ChartFunctions';
import DownloadDetails from './DownloadDetails';

export interface DataBlockProps {
  id: string;
  type: string;
  heading?: string;
  dataBlockRequest?: DataBlockRequest;
  charts?: Chart[];
  tables?: Table[];

  summary?: Summary;

  height?: number;
  showTables?: boolean;
  additionalTabContent?: ReactNode;

  onToggle?: (section: { id: string; title: string }) => void;

  dataBlockResponse?: DataBlockResponse;

  onSummaryDetailsToggle?: (
    isOpened: boolean,
    event: MouseEvent<HTMLElement>,
  ) => void;
}

interface DataBlockState {
  isLoading: boolean;
  isError: boolean;
  charts?: ChartRendererProps[];
  tables?: TableRendererProps[];
  summary?: SummaryRendererProps;
}

class DataBlock extends Component<DataBlockProps, DataBlockState> {
  public static defaultProps = {
    showTables: true,
  };

  public state: DataBlockState = {
    isLoading: false,
    isError: false,
  };

  private query?: DataQuery = undefined;

  public async componentDidMount() {
    const { dataBlockRequest } = this.props;

    this.setState({ isLoading: true, isError: false });

    if (dataBlockRequest) {
      const result = await DataBlockService.getDataBlockForSubject(
        dataBlockRequest,
      );

      if (result) {
        this.parseDataResponse(result);
      } else {
        this.setState({
          isError: true,
          isLoading: false,
        });
      }
    } else {
      const { dataBlockResponse } = this.props;
      if (dataBlockResponse) {
        this.parseDataResponse(dataBlockResponse);
      }
    }
  }

  public async componentWillUnmount() {
    this.query = undefined;
  }

  private parseDataResponse(response: DataBlockResponse): void {
    const newState: DataBlockState = { isLoading: false, isError: false };

    const data: DataBlockData = response;
    const chartMetadata = parseMetaData(response.metaData);

    if (chartMetadata === undefined) return;

    const { charts, summary, tables, heading } = this.props;

    if (response.result.length > 0) {
      if (tables) {
        newState.tables = [
          {
            heading,
            response,
            ...tables[0], /// at present only one chart
          },
        ];
      }
    }

    if (charts) {
      newState.charts = charts.map<ChartRendererProps>(chart => {
        // There is a presumption that the configuration from the API is valid.
        // The data coming from the API is required to be optional for the ChartRenderer
        // But the data for the charts is required. The charts have validation that
        // prevent them from attempting to render.
        // @ts-ignore
        const rendererProps: ChartRendererProps = {
          data,
          meta: chartMetadata,
          ...chart,
        };

        return rendererProps;
      });
    }

    if (summary) {
      newState.summary = {
        ...summary,
        data,
        meta: chartMetadata,
      };
    }
    this.setState(newState);
  }

  public render() {
    const {
      heading,
      height,
      showTables,
      additionalTabContent,
      onToggle,
      onSummaryDetailsToggle,
      id,
    } = this.props;
    const { charts, summary, tables, isLoading, isError } = this.state;
    return (
      <>
        {heading && <h3>{heading}</h3>}

        {isLoading ? (
          <LoadingSpinner text="Loading content..." />
        ) : (
          <Tabs id={id} onToggle={onToggle}>
            {isError && (
              <TabsSection id={`${id}-error`} title="Error">
                An error occurred while loading the data, please try again later
              </TabsSection>
            )}

            {summary && (
              <TabsSection id={`${id}-summary`} title="Summary">
                <SummaryRenderer
                  onToggle={onSummaryDetailsToggle}
                  {...summary}
                />
              </TabsSection>
            )}

            {tables && showTables && (
              <TabsSection id={`${id}-tables`} title="Data tables">
                {tables.map((table, idx) => {
                  const key = `${id}0_table_${idx}`;

                  return (
                    <React.Fragment key={key}>
                      <TimePeriodDataTableRenderer {...table} />
                      <DataSource />
                      <DownloadDetails />
                    </React.Fragment>
                  );
                })}

                {additionalTabContent}
              </TabsSection>
            )}

            {charts && (
              <TabsSection
                datablockId={id}
                id={`${id}-charts`}
                title="Charts"
                lazy={false}
              >
                {charts.map((chart, idx) => {
                  const key = `${id}_chart_${idx}`;

                  return (
                    <React.Fragment key={key}>
                      {chart.data &&
                      chart.meta &&
                      chart.data.result.length > 0 ? (
                        <>
                          <ChartRenderer {...chart} height={height}>
                            <DataSource />
                            <DownloadDetails />
                          </ChartRenderer>
                        </>
                      ) : (
                        <div>
                          Unable to render chart, invalid data configured
                        </div>
                      )}
                    </React.Fragment>
                  );
                })}

                {additionalTabContent}
              </TabsSection>
            )}
          </Tabs>
        )}
      </>
    );
  }
}

export default DataBlock;
