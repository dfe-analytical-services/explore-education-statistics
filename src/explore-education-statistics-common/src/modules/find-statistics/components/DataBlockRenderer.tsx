import LoadingSpinner from '@common/components/LoadingSpinner';
import Tabs from '@common/components/Tabs';
import TabsSection from '@common/components/TabsSection';
import ChartRenderer from '@common/modules/charts/components/ChartRenderer';
import { GetInfographic } from '@common/modules/charts/components/InfographicBlock';
import { AxesConfiguration } from '@common/modules/charts/types/chart';
import getLabelDataSetConfigurations from '@common/modules/charts/util/getLabelDataSetConfigurations';
import useTableQuery from '@common/modules/find-statistics/hooks/useTableQuery';
import TimePeriodDataTable from '@common/modules/table-tool/components/TimePeriodDataTable';
import getDefaultTableHeaderConfig from '@common/modules/table-tool/utils/getDefaultTableHeadersConfig';
import mapTableHeadersConfig from '@common/modules/table-tool/utils/mapTableHeadersConfig';
import { DataBlock } from '@common/services/types/blocks';
import React, { MouseEvent, ReactNode } from 'react';

export interface DataBlockRendererProps {
  additionalTabContent?: ReactNode;
  dataBlock?: DataBlock;
  firstTabs?: ReactNode;
  lastTabs?: ReactNode;
  getInfographic?: GetInfographic;
  id: string;
  onToggle?: (section: { id: string; title: string }) => void;
  onSummaryDetailsToggle?: (
    isOpened: boolean,
    event: MouseEvent<HTMLElement>,
  ) => void;
}

const DataBlockRenderer = ({
  additionalTabContent,
  dataBlock,
  firstTabs,
  lastTabs,
  getInfographic,
  id,
  onToggle,
}: DataBlockRendererProps) => {
  const { value: fullTable, isLoading } = useTableQuery(
    dataBlock
      ? {
          ...dataBlock.dataBlockRequest,
          includeGeoJson: dataBlock.charts.some(chart => chart.type === 'map'),
        }
      : undefined,
  );

  return (
    <LoadingSpinner loading={isLoading}>
      <Tabs id={id} onToggle={onToggle}>
        {firstTabs}

        {dataBlock?.tables?.length && fullTable && (
          <TabsSection id={`${id}-tables`} title="Table">
            {dataBlock?.tables.map((table, index) => {
              return (
                <TimePeriodDataTable
                  key={index}
                  fullTable={fullTable}
                  captionTitle={dataBlock?.heading}
                  source={dataBlock?.source}
                  tableHeadersConfig={
                    table.tableHeaders
                      ? mapTableHeadersConfig(
                          table.tableHeaders,
                          fullTable.subjectMeta,
                        )
                      : getDefaultTableHeaderConfig(fullTable.subjectMeta)
                  }
                />
              );
            })}

            {additionalTabContent}
          </TabsSection>
        )}

        {dataBlock?.charts?.length && fullTable && (
          <TabsSection id={`${id}-charts`} title="Chart">
            <a
              className="govuk-visually-hidden"
              href={`#${id}-tables`}
              aria-live="assertive"
            >
              If you are using a keyboard or a screen reader you may wish to
              view the accessible table instead. Press enter to switch to the
              data tables tab.
            </a>

            {dataBlock?.charts.map((chart, index) => {
              const key = index;

              const axes = { ...chart.axes } as Required<AxesConfiguration>;

              if (
                axes.major?.dataSets?.some(dataSet => !dataSet.config) &&
                chart.labels
              ) {
                axes.major.dataSets = getLabelDataSetConfigurations(
                  chart.labels,
                  axes.major.dataSets,
                );
              }

              if (chart.type === 'infographic') {
                return (
                  <ChartRenderer
                    {...chart}
                    key={key}
                    axes={axes}
                    data={fullTable?.results}
                    meta={fullTable?.subjectMeta}
                    source={dataBlock?.source}
                    getInfographic={getInfographic}
                  />
                );
              }

              return (
                <ChartRenderer
                  {...chart}
                  key={key}
                  axes={axes}
                  data={fullTable?.results}
                  meta={fullTable?.subjectMeta}
                  source={dataBlock?.source}
                />
              );
            })}

            {additionalTabContent}
          </TabsSection>
        )}

        {lastTabs}
      </Tabs>
    </LoadingSpinner>
  );
};

export default DataBlockRenderer;
