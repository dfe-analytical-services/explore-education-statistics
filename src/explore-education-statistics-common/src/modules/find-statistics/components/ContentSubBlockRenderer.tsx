import DataBlock from '@common/modules/find-statistics/components/DataBlock';
import { ContentBlock, Publication } from '@common/services/publicationService';
import ButtonLink from '@common/components/ButtonLink';
import React from 'react';
import ReactMarkdown from 'react-markdown';

export type SectionToggleHandler = (section: {
  id: string;
  title: string;
}) => void;

interface Props {
  block: ContentBlock;
  id: string;
  publication: Publication;
  onToggle?: SectionToggleHandler;
}

const ContentSubBlockRenderer = ({
  block,
  id,
  publication,
  onToggle,
}: Props) => {
  switch (block.type) {
    case 'MarkDownBlock':
      return <ReactMarkdown className="govuk-body" source={block.body} />;
    case 'HtmlBlock':
      return (
        <div
          // eslint-disable-next-line react/no-danger
          dangerouslySetInnerHTML={{ __html: block.body }}
        />
      );
    case 'InsetTextBlock':
      return (
        <div className="govuk-inset-text">
          {block.heading && (
            <h3 className="govuk-heading-s">{block.heading}</h3>
          )}

          <ReactMarkdown className="govuk-body" source={block.body} />
        </div>
      );
    case 'DataBlock':
      return (
        <DataBlock
          {...block}
          id={`${id}_datablock`}
          additionalTabContent={
            <div className="dfe-print-hidden">
              <h2 className="govuk-heading-m">
                Explore and edit this data online
              </h2>
              <p>Use our table tool to explore this data.</p>
              <ButtonLink
                as={`/data-tables/${publication.slug}`}
                to={`/data-tables?publicationSlug=${publication.slug}`}
                href={`/data-tables?publicationSlug=${publication.slug}`}
              >
                Explore data
              </ButtonLink>
            </div>
          }
          onToggle={onToggle}
        />
      );
    default:
      return null;
  }
};

export default ContentSubBlockRenderer;
