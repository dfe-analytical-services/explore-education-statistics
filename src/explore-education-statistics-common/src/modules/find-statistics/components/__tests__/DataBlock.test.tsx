import {
  testChartPropsWithData1,
  testChartPropsWithMultipleData,
  testDataBlockResponse,
} from '@common/modules/charts/components/__tests__/__data__/testBlockData';
import { ChartRendererProps } from '@common/modules/charts/components/ChartRenderer';
import _dataBlockService, {
  DataBlockRequest,
  GeographicLevel,
} from '@common/services/dataBlockService';
import { Chart } from '@common/services/publicationService';
import { render, wait } from '@testing-library/react';
import React from 'react';
import DataBlock from '../DataBlock';

jest.mock('@common/services/dataBlockService');

jest.mock('recharts/lib/util/LogUtils');

const dataBlockService = _dataBlockService as jest.Mocked<
  typeof _dataBlockService
>;

describe('DataBlock', () => {
  const dataBlockRequest: DataBlockRequest = {
    subjectId: '1',
    geographicLevel: GeographicLevel.Country,
    timePeriod: {
      startYear: 2014,
      startCode: 'HT6',
      endYear: 2015,
      endCode: 'HT6',
    },
    filters: ['1', '2'],
    indicators: ['23', '26', '28'],
  };

  test('renders horizontal chart', async () => {
    const getDataBlockForSubject = dataBlockService.getDataBlockForSubject.mockImplementation(
      (_: DataBlockRequest) => {
        return Promise.resolve(testDataBlockResponse);
      },
    );

    const { container } = render(
      <DataBlock
        id="test"
        type="datablock"
        dataBlockRequest={dataBlockRequest}
        showTables={false}
        charts={[
          {
            ...testChartPropsWithData1,
            type: 'horizontalbar',
          } as Chart,
        ]}
      />,
    );

    await wait();

    expect(getDataBlockForSubject).toBeCalledWith(dataBlockRequest);

    expect(
      container.querySelectorAll('section.govuk-tabs__panel'),
    ).toHaveLength(1);

    expect(container.querySelectorAll('.recharts-bar')).toHaveLength(3);
  });

  test('renders vertical chart', async () => {
    const getDataBlockForSubject = dataBlockService.getDataBlockForSubject.mockImplementation(
      (_: DataBlockRequest) => {
        return Promise.resolve(testDataBlockResponse);
      },
    );

    const { container } = render(
      <DataBlock
        id="test"
        type="datablock"
        dataBlockRequest={dataBlockRequest}
        showTables={false}
        charts={[
          {
            ...testChartPropsWithData1,
            type: 'verticalbar',
          } as Chart,
        ]}
      />,
    );

    await wait();

    expect(getDataBlockForSubject).toBeCalledWith(dataBlockRequest);

    expect(
      container.querySelectorAll('section.govuk-tabs__panel'),
    ).toHaveLength(1);

    expect(container.querySelectorAll('.recharts-bar')).toHaveLength(3);
  });

  test('renders table', () => {
    const getDataBlockForSubject = dataBlockService.getDataBlockForSubject.mockImplementation(
      (_: DataBlockRequest) => {
        return Promise.resolve(testDataBlockResponse);
      },
    );

    const { container } = render(
      <DataBlock
        id="test"
        type="datablock"
        dataBlockRequest={dataBlockRequest}
        showTables
      />,
    );

    expect(getDataBlockForSubject).toBeCalledWith(dataBlockRequest);

    expect(container.querySelector('#test-datablock-table')).toMatchSnapshot();
  });

  test('renders summary', async () => {
    const getDataBlockForSubject = dataBlockService.getDataBlockForSubject.mockImplementation(
      (_: DataBlockRequest) => {
        return Promise.resolve(testDataBlockResponse);
      },
    );

    const { container } = render(
      <DataBlock
        id="test-datablock"
        type="databock"
        dataBlockRequest={dataBlockRequest}
        showTables={false}
        summary={{
          dataKeys: ['23', '26', '28'],
          dataSummary: ['up 10%', 'down 10%', 'up 11%'],
          dataDefinition: ['a', 'b', 'c'],
          dataDefinitionTitle: ['a', 'b', 'c'],
        }}
      />,
    );

    await wait();

    expect(getDataBlockForSubject).toBeCalledWith(dataBlockRequest);

    expect(
      container.querySelector('#test-datablock-summary'),
    ).toMatchSnapshot();
  });

  test('renders map instead of chart', async () => {
    const getDataBlockForSubject = dataBlockService.getDataBlockForSubject.mockImplementation(
      (_: DataBlockRequest) => {
        return Promise.resolve(testDataBlockResponse);
      },
    );

    const { container } = render(
      <DataBlock
        id="test-datablock"
        type="datablock"
        dataBlockRequest={dataBlockRequest}
        showTables={false}
        charts={[
          {
            ...testChartPropsWithMultipleData,
            type: 'map',
          } as ChartRendererProps,
        ]}
      />,
    );

    await wait();

    expect(getDataBlockForSubject).toBeCalledWith(dataBlockRequest);

    expect(container.querySelector('#test-datablock-charts')).toMatchSnapshot();
  });
});
