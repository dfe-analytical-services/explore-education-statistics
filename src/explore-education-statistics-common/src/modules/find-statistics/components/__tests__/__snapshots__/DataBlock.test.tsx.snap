// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataBlock renders map instead of chart 1`] = `
<section
  aria-labelledby="test-datablock-charts-tab"
  class="govuk-tabs__panel dfe-content-overflow panel"
  id="test-datablock-charts"
  role="tabpanel"
  tabindex="-1"
>
  
  <a
    aria-live="assertive"
    class="govuk-visually-hidden"
    href="#test-datablock-tables"
  >
    If you are using a keyboard or a screen reader you may wish to view the accessible table instead. Press enter to switch to the data tables tab.
  </a>
  <form>
    <div
      class="govuk-form-group"
    >
      <fieldset
        class="govuk-fieldset"
        id="data-and-location"
      >
        <legend
          class="govuk-fieldset__legend govuk-fieldset__legend--l govuk-visually-hidden"
        >
          Select data and location to view indicators
        </legend>
        <div
          class="mapContainer"
        >
          <div
            class="govuk-form-group mapIndicator"
          >
            <label
              class="govuk-label"
              for="selectedIndicator"
            >
              1. Select data to view
            </label>
            <select
              class="govuk-select govuk-!-width-full"
              id="selectedIndicator"
              name="selectedIndicator"
            >
              <option
                value="23_1_2_____"
              >
                Label 23_1_2
              </option>
              <option
                value="26_1_2_____"
              >
                Label 26_1_2
              </option>
              <option
                value="28_1_2_____"
              >
                Label 28_1_2
              </option>
            </select>
          </div>
          <div
            class="govuk-form-group mapLocation"
          >
            <label
              class="govuk-label"
              for="selectedLocation"
            >
              2. Select a location
            </label>
            <select
              class="govuk-select"
              id="selectedLocation"
              name="selectedLocation"
            >
              <option
                value=""
              >
                Select...
              </option>
              <option
                value="E92000001"
              >
                England
              </option>
              <option
                value="S92000001"
              >
                Scotland
              </option>
            </select>
          </div>
        </div>
      </fieldset>
    </div>
  </form>
  <div
    class="mapContainer"
  >
    <div
      class="mapView"
    >
      <div
        class="map dfe-print-break-avoid leaflet-container leaflet-grab leaflet-touch-drag"
        style="width: 800px; height: 600px; position: relative;"
        tabindex="0"
      >
        <div
          class="leaflet-pane leaflet-map-pane"
          style="left: 0px; top: 0px;"
        >
          <div
            class="leaflet-pane leaflet-tile-pane"
          />
          <div
            class="leaflet-pane leaflet-shadow-pane"
          />
          <div
            class="leaflet-pane leaflet-overlay-pane"
          >
            <svg
              height="0"
              pointer-events="none"
              style="left: 0px; top: 0px;"
              viewBox="0 0 0 0"
              width="0"
            >
              <g>
                <path
                  class="uk leaflet-interactive"
                  d="M-3 3L-3 -3L3 -3L3 3z"
                  fill="#3388ff"
                  fill-opacity="0.2"
                  fill-rule="evenodd"
                  stroke="#3388ff"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-opacity="1"
                  stroke-width="3"
                />
                <path
                  class="leaflet-interactive"
                  d="M0 0"
                  fill="#3f0000"
                  fill-opacity="0.2"
                  fill-rule="evenodd"
                  stroke="#3388ff"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-opacity="1"
                  stroke-width="3"
                />
              </g>
            </svg>
          </div>
          <div
            class="leaflet-pane leaflet-marker-pane"
          />
          <div
            class="leaflet-pane leaflet-tooltip-pane"
          />
          <div
            class="leaflet-pane leaflet-popup-pane"
          />
        </div>
        <div
          class="leaflet-control-container"
        >
          <div
            class="leaflet-top leaflet-left"
          >
            <div
              class="leaflet-control-zoom leaflet-bar leaflet-control"
            >
              <a
                aria-label="Zoom in"
                class="leaflet-control-zoom-in"
                href="#"
                role="button"
                title="Zoom in"
              >
                +
              </a>
              <a
                aria-label="Zoom out"
                class="leaflet-control-zoom-out"
                href="#"
                role="button"
                title="Zoom out"
              >
                −
              </a>
            </div>
          </div>
          <div
            class="leaflet-top leaflet-right"
          />
          <div
            class="leaflet-bottom leaflet-left"
          />
          <div
            class="leaflet-bottom leaflet-right"
          >
            <div
              class="leaflet-control-attribution leaflet-control"
            >
              <a
                href="https://leafletjs.com"
                title="A JS library for interactive maps"
              >
                Leaflet
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      aria-live="assertive"
      class="mapControls"
    >
      <div
        class="mapKey"
      >
        <h3
          class="govuk-heading-s"
        >
          Key to 
          Label 23_1_2
        </h3>
        <dl
          class="govuk-list"
        >
          <dd
            class="legend"
          >
            <span
              class="rate0"
              style="background-color: rgb(63, 0, 0);"
            >
               
            </span>
             
            -3
            %
              to 
            -3
            %
             
          </dd>
          <dd
            class="legend"
          >
            <span
              class="rate1"
              style="background-color: rgb(111, 0, 0);"
            >
               
            </span>
             
            -3
            %
              to 
            -3
            %
             
          </dd>
          <dd
            class="legend"
          >
            <span
              class="rate2"
              style="background-color: rgb(159, 0, 0);"
            >
               
            </span>
             
            -3
            %
              to 
            -2
            %
             
          </dd>
          <dd
            class="legend"
          >
            <span
              class="rate3"
              style="background-color: rgb(207, 0, 0);"
            >
               
            </span>
             
            -2
            %
              to 
            -2
            %
             
          </dd>
          <dd
            class="legend"
          >
            <span
              class="rate4"
              style="background-color: rgb(255, 0, 0);"
            >
               
            </span>
             
            -2
            %
              to 
            -2
            %
             
          </dd>
        </dl>
      </div>
    </div>
  </div>
  <div
    class="govuk-!-margin-top-6"
  >
    <p
      class="govuk-body-s"
    >
      Source: Department for Education
    </p>
  </div>
</section>
`;

exports[`DataBlock renders summary 1`] = `
<section
  aria-labelledby="test-datablock-summary-tab"
  class="govuk-tabs__panel dfe-content-overflow panel"
  id="test-datablock-summary"
  role="tabpanel"
  tabindex="-1"
>
  
  <div
    class="keyStatsContainer"
  >
    <div
      class="keyStatTile"
    >
      <div
        class="keyStat"
      >
        <h3
          class="govuk-heading-s"
          data-testid="key-stat-tile-title"
        >
          Unauthorised absence rate
        </h3>
        <p
          class="govuk-heading-xl"
          data-testid="key-stat-tile-value"
        >
          -3%
        </p>
        <p
          class="govuk-body-s"
        >
          up 10%
        </p>
      </div>
      <details
        class="govuk-details"
        role="group"
      >
        <summary
          aria-controls="details-content-4"
          aria-expanded="false"
          class="govuk-details__summary"
          role="button"
          tabindex="0"
        >
          <span
            class="govuk-details__summary-text"
            data-testid="Expand Details Section Define 'Unauthorised absence rate'"
          >
            Define 'Unauthorised absence rate'
          </span>
        </summary>
        <div
          aria-hidden="true"
          class="govuk-details__text"
          id="details-content-4"
        >
          <div>
            a
          </div>
        </div>
      </details>
    </div>
    <div
      class="keyStatTile"
    >
      <div
        class="keyStat"
      >
        <h3
          class="govuk-heading-s"
          data-testid="key-stat-tile-title"
        >
          Overall absence rate
        </h3>
        <p
          class="govuk-heading-xl"
          data-testid="key-stat-tile-value"
        >
          4%
        </p>
        <p
          class="govuk-body-s"
        >
          down 10%
        </p>
      </div>
      <details
        class="govuk-details"
        role="group"
      >
        <summary
          aria-controls="details-content-5"
          aria-expanded="false"
          class="govuk-details__summary"
          role="button"
          tabindex="0"
        >
          <span
            class="govuk-details__summary-text"
            data-testid="Expand Details Section Define 'Overall absence rate'"
          >
            Define 'Overall absence rate'
          </span>
        </summary>
        <div
          aria-hidden="true"
          class="govuk-details__text"
          id="details-content-5"
        >
          <div>
            b
          </div>
        </div>
      </details>
    </div>
    <div
      class="keyStatTile"
    >
      <div
        class="keyStat"
      >
        <h3
          class="govuk-heading-s"
          data-testid="key-stat-tile-title"
        >
          Authorised absence rate
        </h3>
        <p
          class="govuk-heading-xl"
          data-testid="key-stat-tile-value"
        >
          1%
        </p>
        <p
          class="govuk-body-s"
        >
          up 11%
        </p>
      </div>
      <details
        class="govuk-details"
        role="group"
      >
        <summary
          aria-controls="details-content-6"
          aria-expanded="false"
          class="govuk-details__summary"
          role="button"
          tabindex="0"
        >
          <span
            class="govuk-details__summary-text"
            data-testid="Expand Details Section Define 'Authorised absence rate'"
          >
            Define 'Authorised absence rate'
          </span>
        </summary>
        <div
          aria-hidden="true"
          class="govuk-details__text"
          id="details-content-6"
        >
          <div>
            c
          </div>
        </div>
      </details>
    </div>
  </div>
  &lt;div&gt;test&lt;/div&gt;
</section>
`;

exports[`DataBlock renders table 1`] = `null`;
